jsx.js

```diff
@@ -13,7 +13,78 @@ module.exports = {
   },
 
   rules: {
-    'jsx-quotes': [2, 'prefer-double']
+    'jsx-quotes': [2, 'prefer-double'],
+    'react/boolean-prop-naming': [0],
+    'react/button-has-type': [0],
+    'react/default-props-match-prop-types': [2],
+    'react/destructuring-assignment': [0],
+    'react/forbid-component-props': [0],
+    'react/forbid-dom-props': [0],
+    'react/forbid-elements': [0],
+    'react/forbid-prop-types': [2],
+    'react/function-component-definition': [2, {namedComponents: 'arrow-function'}],
+    'react/jsx-boolean-value': [0],
+    'react/jsx-child-element-spacing': [2],
+    'react/jsx-closing-bracket-location': [2, 'line-aligned'],
+    'react/jsx-closing-tag-location': [2],
+    'react/jsx-curly-brace-presence': [2, {props: 'never', children: 'never'}],
+    'react/jsx-curly-newline': [2],
+    'react/jsx-curly-spacing': [2, 'never'],
+    'react/jsx-equals-spacing': [0],
+    'react/jsx-filename-extension': [0],
+    'react/jsx-first-prop-new-line': [2],
+    'react/jsx-fragments': [2, 'syntax'],
+    'react/jsx-handler-names': [0],
+    'react/jsx-indent': [2, 2, {checkAttributes: true, indentLogicalExpressions: true}],
+    'react/jsx-indent-props': [2, 2],
+    'react/jsx-max-depth': [0],
+    'react/jsx-max-props-per-line': [2],
+    'react/jsx-newline': [2, {prevent: true}],
+    'react/jsx-no-constructed-context-values': [2],
+    'react/jsx-no-literals': [0],
+    'react/jsx-no-script-url': [0],
+    'react/jsx-no-useless-fragment': [2, {allowExpressions: true}],
+    'react/jsx-one-expression-per-line': [2, {allow: 'literal'}],
+    'react/jsx-pascal-case': [2],
+    'react/jsx-props-no-multi-spaces': [2],
+    'react/jsx-props-no-spreading': [0],
+    'react/jsx-sort-default-props': [0],
+    'react/jsx-sort-props': [0],
+    'react/jsx-tag-spacing': [2, {beforeClosing: 'never'}],
+    'react/jsx-wrap-multilines': [2, {
+      declaration: 'parens-new-line',
+      assignment: 'parens-new-line',
+      return: 'parens-new-line',
+      arrow: 'parens-new-line',
+      condition: 'parens-new-line',
+      logical: 'parens-new-line',
+      prop: 'parens-new-line'
+    }],
+    'react/no-access-state-in-setstate': [2],
+    'react/no-adjacent-inline-elements': [0],
+    'react/no-array-index-key': [2],
```

Comment:
This is a good rule in principle, but often we just don't have anything better to index on, and I suspect we'd have to disable this rule more often than not.

Reply:
could we add it as a warning perhaps?

Reply:
> often we just don't have anything better to index on

I feel like it's quite rare to display items from an array where no combination of a subset of properties from the array item uniquely identifies it. I've encountered many usages of keying by the array index where it seems like the developer just didn't put enough thought into what the identifying properties of the array items are.

In cases where it's really not possible to uniquely identify array items using a subset of their properties, I find that it is indicative of a common anti-pattern, where the developer should just have used plain JSX, but "abstracted away" the duplicate code by using an array.

Reply:
To be clear: I'm in favour of making this rule an error.

Reply:
> could we add it as a warning perhaps?

The only rules we should make warnings of are things that we are happy to be piling up on `main`. Anything else should be an error or off.